app/src/main/java/org/simple/clinic/widgets/BindingPagingItemAdapter.kt

```diff
@@ -0,0 +1,50 @@
+package org.simple.clinic.widgets
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.paging.PagedListAdapter
+import androidx.recyclerview.widget.DiffUtil
+import androidx.viewbinding.ViewBinding
+import io.reactivex.Observable
+import io.reactivex.subjects.PublishSubject
+import io.reactivex.subjects.Subject
+import org.simple.clinic.widgets.recyclerview.BindingViewHolder
+
+// TODO (SM) : Rename to PagingItemAdapter once migration to ViewBinding is finished
+open class BindingPagingItemAdapter<I : BindingPagingItemAdapter.Item<E>, E>(
+    diffCallback: DiffUtil.ItemCallback<I>,
+    private val bindings: Map<Int, (layoutInflater: LayoutInflater, parent: ViewGroup) -> ViewBinding>
```

Comment:
What do you think of extracting the type `(layoutInflater: LayoutInflater, parent: ViewGroup) -> ViewBinding` to a type alias for better readability?

Reply:
Yeah, good point. Will do that